src/app/core/http-services/item-navigation.service.ts

```diff
@@ -199,4 +199,45 @@ export class ItemNavigationService {
     return (isSkill(type)) ? this.getRootSkills() : this.getRootActivities();
   }
 
+  getNavigationNeighbors(itemRoute: ItemRoute): Observable<{ parent: ItemRoute|null, left: ItemRoute|null, right: ItemRoute|null }> {
+    const parentId = itemRoute.path[itemRoute.path.length - 1];
+
+    // Root activity => no parent/left/right activity
+    if (!parentId) return of({ parent: null, left: null, right: null });
+
+    return this.getNavDataFromChildRoute(parentId, itemRoute).pipe(map(navParent => {
+      const index = navParent.items.findIndex(item => item.id === itemRoute.id);
+      if (index === -1) throw new Error(`Unexpected: item is missing from its parent children list`);
+
+      const parentAttemptId = navParent.parent.attemptId;
+      if (parentAttemptId === null) throw new Error('Unexpected: parent of an item node has no attempt');
+
+      const typeCat : ItemTypeCategory = itemRoute.contentType === 'activity' ? 'activity' : 'skill';
+
+      const leftItem = navParent.items[index - 1];
+      const left: ItemRoute|null = leftItem ? {
+        id: leftItem.id,
+        contentType: typeCat,
+        ...(leftItem.attemptId ? { attemptId: leftItem.attemptId } : { parentAttemptId }),
+        path: itemRoute.path,
+      } : null;
+
+      const rightItem = navParent.items[index + 1];
+      const right: ItemRoute|null = rightItem ? {
+        id: rightItem.id,
+        contentType: typeCat,
+        ...(rightItem.attemptId ? { attemptId: rightItem.attemptId } : { parentAttemptId }),
+        path: itemRoute.path,
+      } : null;
+
+      const parent: ItemRoute = {
+        id: parentId,
+        contentType: typeCat,
+        path: itemRoute.path.slice(0, -1),
+        attemptId: isRouteWithAttempt(itemRoute) ? itemRoute.attemptId : itemRoute.parentAttemptId,
```

Comment:
For the attempt id of the parent, it is not the one of its child (`itemRoute.attemptId`). But you get it from the service directly `parentAttemptId` that you already use, but the parent, it is its own attempt id.

Reply:
fixed